packages/nuxt/src/plugins/plugin.ts

```diff
@@ -0,0 +1,284 @@
+    if (
+      !siblingFiles.includes('package.json') &&
+      !siblingFiles.includes('project.json')
+    ) {
+      return {};
+    }
+
+    options = normalizeOptions(options);
+
+    const hash = calculateHashForCreateNodes(projectRoot, options, context);
+    const targets = targetsCache[hash]
+      ? targetsCache[hash]
+      : await buildNuxtTargets(configFilePath, projectRoot, options, context);
+
+    calculatedTargets[hash] = targets;
+
+    return {
+      projects: {
+        [projectRoot]: {
+          root: projectRoot,
+          targets,
+        },
+      },
+    };
+  },
+];
+
+async function buildNuxtTargets(
+  configFilePath: string,
+  projectRoot: string,
+  options: NuxtPluginOptions,
+  context: CreateNodesContext
+) {
+  const nuxtConfig: {
+    buildDir: string;
+  } = await getInfoFromNuxtConfig(configFilePath, context, projectRoot);
+
+  const { buildOutputs, testOutputs } = getOutputs(projectRoot, nuxtConfig);
+
+  const namedInputs = getNamedInputs(projectRoot, context);
+
+  const targets: Record<string, TargetConfiguration> = {};
+
+  targets[options.buildTargetName] = buildTarget(
+    context,
+    namedInputs,
+    buildOutputs,
+    projectRoot
+  );
+
+  targets[options.serveTargetName] = serveTarget();
+
+  targets[options.testTargetName] = testTarget(
+    context,
+    namedInputs,
+    testOutputs,
+    options,
+    projectRoot
+  );
+
+  return targets;
+}
+
+function buildTarget(
+  context: CreateNodesContext,
+  namedInputs: {
+    [inputName: string]: any[];
+  },
+  buildOutputs: string[],
+  projectRoot: string
+) {
+  const targetDefaults = readTargetDefaultsForTarget(
+    'build',
+    context.nxJsonConfiguration.targetDefaults,
+    '@nx/nuxt:build'
+  );
+
+  const targetConfig: TargetConfiguration = {
+    command: `${
+      getPackageManagerCommand(detectPackageManager()).exec
+    } nuxi build ${projectRoot}`,
+    // this seems to not have any effect
+    // options: {
+    //   cwd: joinPathFragments(projectRoot),
+    // },
+  };
+
+  if (targetDefaults?.outputs === undefined) {
+    targetConfig.outputs = buildOutputs;
+  }
+
+  if (targetDefaults?.cache === undefined) {
+    targetConfig.cache = true;
+  }
+
+  if (targetDefaults?.inputs === undefined) {
+    targetConfig.inputs =
+      'production' in namedInputs
+        ? ['production', '^production']
+        : ['default', '^default'];
+  }
+
+  return targetConfig;
+}
+
+function serveTarget() {
+  const targetConfig: TargetConfiguration = {
+    executor: '@nx/nuxt:serve',
+    defaultConfiguration: 'development',
+  };
+
+  return targetConfig;
+}
+
+function testTarget(
+  context: CreateNodesContext,
+  namedInputs: {
+    [inputName: string]: any[];
+  },
+  outputs: string[],
+  options: NuxtPluginOptions,
+  projectRoot: string
+) {
+  const targetDefaults = readTargetDefaultsForTarget(
+    options.testTargetName,
+    context.nxJsonConfiguration.targetDefaults,
+    '@nx/vite:test'
```

Comment:
We don't need read from `@nx/vite:test` for defaults I think, since we're not using the executor here.